src/lib.rs

```diff
@@ -0,0 +1,94 @@
+mod clock;
+pub mod lnd;
+mod onion_messenger;
+mod rate_limit;
+
+use crate::lnd::{
+    features_support_onion_messages, get_lnd_client, string_to_network, LndCfg, LndNodeSigner,
+};
+
+use crate::onion_messenger::{run_onion_messenger, MessengerUtilities};
+use bitcoin::secp256k1::PublicKey;
+use lightning::ln::peer_handler::IgnoringMessageHandler;
+use lightning::onion_message::{DefaultMessageRouter, OnionMessenger};
+use log::{error, info};
+use std::collections::HashMap;
+use std::str::FromStr;
+use tonic_lnd::lnrpc::GetInfoRequest;
+
+pub async fn run(args: LndCfg) -> Result<(), ()> {
```

Comment:
(Pre-existing) I think that we need to do a more general errors followup, so not for this PR but I think that this should return a defined error.